code/modules/mob/living/simple_animal/borer/borer.dm

```diff
@@ -0,0 +1,799 @@
+				host_brain.lastKnownIP = h2b_ip
+
+			// self -> host
+			var/s2h_id = src.computer_id
+			var/s2h_ip= src.lastKnownIP
+			src.computer_id = null
+			src.lastKnownIP = null
+
+			victim.ckey = src.ckey
+			victim.mind = src.mind
+
+			if(!victim.computer_id)
+				victim.computer_id = s2h_id
+
+			if(!victim.lastKnownIP)
+				victim.lastKnownIP = s2h_ip
+
+			controlling = 1
+
+			victim.verbs += /mob/living/carbon/proc/release_control
+			victim.verbs += /mob/living/carbon/proc/spawn_larvae
+			victim.verbs -= /mob/living/proc/borer_comm
+			victim.verbs += /mob/living/proc/trapped_mind_comm
+
+			victim.med_hud_set_status()
+
+/mob/living/simple_animal/borer/verb/punish()
+	set category = "Borer"
+	set name = "Punish"
+	set desc = "Punish your victim."
+
+	if(!victim)
+		src << "<span class='warning'>You are not inside a host body.</span>"
+		return
+
+	if(src.stat != CONSCIOUS)
+		src << "You cannot do that in your current state."
+		return
+
+	if(docile)
+		src << "<span class='warning'>You are feeling far too docile to do that.</span>"
+		return
+
+	if(chemicals < 75)
+		src << "<span class='warning'>You need 75 chems to punish your host.</span>"
+		return
+
+	var/punishment = input("Select a punishment:.", "Punish") as null|anything in list("Blindness","Deafness","Stun")
+
+	if(!punishment)
+		return
+
+	if(chemicals < 75)
+		src << "<span class='warning'>You need 75 chems to punish your host.</span>"
+		return
+
+	switch(punishment) //Hardcoding this stuff.
+		if("Blindness")
+			victim.blind_eyes(2)
+		if("Deafness")
+			victim.ear_deaf = 20
+		if("Stun")
+			victim.Weaken(10)
+
+	log_game("[src]/([src.ckey]) punished [victim]/([victim.ckey] with [punishment]")
+
+	chemicals -= 75
+
+
+mob/living/carbon/proc/release_control()
+
+	set category = "Borer"
+	set name = "Release Control"
+	set desc = "Release control of your host's body."
+
+	if(borer && borer.host_brain)
+		src << "<span class='danger'>You withdraw your probosci, releasing control of [borer.host_brain]</span>"
+
+		borer.detatch()
+
+		verbs -= /mob/living/carbon/proc/release_control
+		verbs -= /mob/living/carbon/proc/spawn_larvae
+		verbs += /mob/living/proc/borer_comm
+		verbs -= /mob/living/proc/trapped_mind_comm
+
+//Check for brain worms in head.
+/mob/proc/has_brain_worms()
+
+	for(var/I in contents)
+		if(istype(I,/mob/living/simple_animal/borer))
+			return I
+
+	return 0
+
+/mob/living/carbon/proc/spawn_larvae()
+	set category = "Borer"
+	set name = "Reproduce"
+	set desc = "Spawn several young."
+
+	var/mob/living/simple_animal/borer/B = has_brain_worms()
+
+	if(istype(src, /mob/living/brain))
+		src << "<span class='usernotice'>You need a mouth to be able to do this.</span>"
+		return
+	if(!B)
+		return
+
+	if(B.chemicals >= 100)
+		visible_message("<span class='danger'>[src] heaves violently, expelling a rush of vomit and a wriggling, sluglike creature!</span>")
+		B.chemicals -= 100
+
+		new /obj/effect/decal/cleanable/vomit(get_turf(src))
```

Comment:
you should get_turf once and store it in a var, that way you can also check that get_turf returns something, so you don't spawn decals or mobs into nullspace.